hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/snapshot/OMSnapshotMoveDeletedKeysResponse.java

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.hadoop.ozone.om.response.snapshot;
+
+import org.apache.hadoop.hdds.utils.db.BatchOperation;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.OmSnapshot;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.RepeatedOmKeyInfo;
+import org.apache.hadoop.ozone.om.response.CleanupTableInfo;
+import org.apache.hadoop.ozone.om.response.OMClientResponse;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.KeyInfo;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.SnapshotMoveKeyInfos;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMResponse;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.List;
+
+import static org.apache.hadoop.ozone.om.OmMetadataManagerImpl.SNAPSHOT_INFO_TABLE;
+
+/**
+ * Response for OMSnapshotMoveDeletedKeysRequest.
+ */
+@CleanupTableInfo(cleanupTables = {SNAPSHOT_INFO_TABLE})
+public class OMSnapshotMoveDeletedKeysResponse extends OMClientResponse {
+
+  private OmSnapshot nextSnapshot;
+  private List<SnapshotMoveKeyInfos> activeDBKeysList;
+  private List<SnapshotMoveKeyInfos> nextDBKeysList;
+
+  public OMSnapshotMoveDeletedKeysResponse(OMResponse omResponse,
+       OmSnapshot omNextSnapshot, List<SnapshotMoveKeyInfos> activeDBKeysList,
+       List<SnapshotMoveKeyInfos> nextDBKeysList) {
+    super(omResponse);
+    this.nextSnapshot = omNextSnapshot;
+    this.activeDBKeysList = activeDBKeysList;
+    this.nextDBKeysList = nextDBKeysList;
+  }
+
+  /**
+   * For when the request is not successful.
+   * For a successful request, the other constructor should be used.
+   */
+  public OMSnapshotMoveDeletedKeysResponse(@Nonnull OMResponse omResponse) {
+    super(omResponse);
+    checkStatusNotOK();
+  }
+
+  @Override
+  protected void addToDBBatch(OMMetadataManager omMetadataManager,
+      BatchOperation batchOperation) throws IOException {
+
+    for (SnapshotMoveKeyInfos activeDBKey : activeDBKeysList) {
+      RepeatedOmKeyInfo activeDBOmKeyInfo =
+          createRepeatedOmKeyInfo(activeDBKey.getKeyInfosList());
+
+      if (activeDBOmKeyInfo == null) {
+        continue;
+      }
+
+      omMetadataManager.getDeletedTable().putWithBatch(
+          batchOperation, activeDBKey.getKey(), activeDBOmKeyInfo);
+    }
+
+    for (SnapshotMoveKeyInfos nextDBKey : nextDBKeysList) {
+      RepeatedOmKeyInfo nextDBOmKeyInfo =
+          createRepeatedOmKeyInfo(nextDBKey.getKeyInfosList());
+
+      if (nextDBOmKeyInfo == null) {
+        continue;
+      }
+
+      if (nextSnapshot != null) {
+        nextSnapshot.getMetadataManager()
+            .getDeletedTable().putWithBatch(batchOperation,
```

Comment:
Is it a problem that we are doing the put with the snapshots metadata manager, but the commit is happening to the omMetadataManage?r: https://github.com/apache/ozone/blob/3de525c60c40441a938f84ffda61ebd2b8bad50b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java#L346-L347

Reply:
You are correct. I have updated the patch to separate batch operation for snapshots.